File: packages/cloud-adapters/secrets/interface.ts
// Secrets adapter axis — KMS-style envelope ops the platform uses for
// per-tenant data encryption. Crypto-shred = key delete (Discussion #76).
// Bare-metal is Vault Transit; cloud impls map to AWS Secrets Manager /
// GCP Secret Manager / Azure Key Vault.

export interface SecretsAdapter {
  mintKey(name: string): Promise<void>;
  encrypt(name: string, plaintext: Uint8Array): Promise<string>;
  decrypt(name: string, ciphertext: string): Promise<Uint8Array>;
  rotateKey(name: string): Promise<void>;
  deleteKey(name: string): Promise<void>;
}

File: packages/cloud-adapters/secrets/interface.go
// Package secrets defines the cloud-adapter secrets axis interface.
// Bare-metal impl: Vault Transit. Cloud impls: AWS Secrets Manager /
// GCP Secret Manager / Azure Key Vault.
package secrets

import "context"

type Adapter interface {
	MintKey(ctx context.Context, name string) error
	Encrypt(ctx context.Context, name string, plaintext []byte) (string, error)
	Decrypt(ctx context.Context, name string, ciphertext string) ([]byte, error)
	RotateKey(ctx context.Context, name string) error
	DeleteKey(ctx context.Context, name string) error
}
